matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/DefaultTimeline.kt

```diff
@@ -516,11 +516,17 @@ internal class DefaultTimeline(
     }
 
     /**
-     * This has to be called on TimelineThread as it access realm live results
+     * This has to be called on TimelineThread as it accesses realm live results
      */
     private fun executePaginationTask(direction: Timeline.Direction, limit: Int) {
         val token = getTokenLive(direction)
         if (token == null) {
+            val currentChunk = getLiveChunk()
+            if (direction == Timeline.Direction.FORWARDS && currentChunk?.hasBeenALastForwardChunk() == true) {
+                // We are in the case that next event exists, but we do not know the next token.
+                // Fetch (again) the last event to get a nextToken
+                currentChunk.timelineEvents.lastOrNull()?.eventId?.let { fetchEvent(it) }
```

Comment:
Don't get lastOrNull on chunk as it's not sorted. I think you should use the nonFilteredEvents instead and take the first which is the most recent.